app/models/thread.php

```diff
@@ -60,15 +62,113 @@ public function create(Comment $comment)
         $db->commit();
     }
 
-    public function getThreadInfo()
+    public static function getThreadInfo($id = NULL)
     {
+        if (empty($id)) {
+            return false;
+        }
         $db = DB::conn();
-        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($this->id));
-        $thread_username = $db->row('SELECT username FROM user WHERE id = ?', array($row['user_id']));
-        return array(
-            'username' => $thread_username['username'],
-            'date' => $row['created'],
-            'user_id' => $row['user_id']
+        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
+        $thread_username = $db->row('SELECT username,avatar FROM user WHERE id = ?', array($row['user_id']));
+        $num_comment = $db->value('SELECT COUNT(*) FROM comment WHERE thread_id = ?', array($id));
+        $num_follow = $db->value('SELECT COUNT(*) FROM follow WHERE thread_id = ?', array($id));
```

Comment:
same principle as this comment: https://github.com/j-asis/Web-Portal/pull/2/files#r27791552 :bow: